Carta.py

```diff
@@ -320,33 +341,73 @@ def opponentRespondsInGrabPhase(self):
         if ((self.GPinfo.yourGrabCardLastWord == self.curReadingCard.getLastWord()) and \
             (self.GPinfo.oppoGrabCardLastWord == self.curReadingCard.getLastWord()) and \
             (self.GPinfo.yourTime < self.GPinfo.opponentTime)):
             self.GPinfo.youWin = True
             self.GPinfo.opponentWin = False
             statement = "Carta.py: decideWinner: Both players grabbed the right card, but you grabbed it faster than opponent"
+            self.dialogs.append("You Win, Opponent Loses")
+            if (self.GPinfo.yourGrabCardSide == GrabbingCardSide.YOUR_SIDE):
+                self.numYourGrabCardInPlay -= 1
+            elif (self.GPinfo.yourGrabCardSide == GrabbingCardSide.OPPONENT_SIDE):
+                self.numOppoGrabCardInPlay -= 1
+                self.phase = Phase.YOUR_ADJUSTMENT
+                self.numCardsToTransfer += 1
         # Both players grabbed the right card, but opponent grabbed it faster than you
         elif ((self.GPinfo.yourGrabCardLastWord == self.curReadingCard.getLastWord()) and \
               (self.GPinfo.oppoGrabCardLastWord == self.curReadingCard.getLastWord()) and \
               (self.GPinfo.yourTime > self.GPinfo.opponentTime)):
             self.GPinfo.youWin = False
             self.GPinfo.opponentWin = True
             statement = "Carta.py: decideWinner: Both players grabbed the right card, but opponent grabbed it faster than you"
-        # you didn't grab the right card, but opponent grabbed the right card
+            self.dialogs.append("Opponent Wins, You Lose")
+            self.shouldOpponentTransfer(False)
+        # you grabbed the wrong card, but opponent grabbed the right card
         elif ((self.GPinfo.yourGrabCardLastWord != self.curReadingCard.getLastWord()) and \
               (self.GPinfo.oppoGrabCardLastWord == self.curReadingCard.getLastWord())):
             self.GPinfo.youWin = False
-            statement = "Carta.py: decideWinner: you didn't grab the right card, but opponent grabbed the right card"
+            self.GPinfo.opponentWin = True
+            statement = "Carta.py: decideWinner: you grabbed the wrong card, but opponent grabbed the right card"
+            self.dialogs.append("Opponent Wins, You Lose")
+            self.shouldOpponentTransfer(True)
         # you grabbed the right card, but opponent didn't grab the right card
         elif ((self.GPinfo.yourGrabCardLastWord == self.curReadingCard.getLastWord()) and \
               (self.GPinfo.oppoGrabCardLastWord != self.curReadingCard.getLastWord())):
             self.GPinfo.youWin = True
             self.GPinfo.opponentWin = False
             statement = "Carta.py: decideWinner: you grabbed the right card, but opponent didn't grab the right card"
+            self.dialogs.append("You Win , Opponent Loses")
+            if (self.GPinfo.yourGrabCardSide == GrabbingCardSide.YOUR_SIDE):
+                self.numYourGrabCardInPlay -= 1
```

Comment:
Change about `self.numYourGrabCardInPlay` or the one for opponent

should be considered totally (after transfer phase)

Example, your true touch, opponent true touch, you are faster:

result:

`numYourGrabCards -= 1`. Nothing about opponent. No matter you need to adjust or not.

Reply:
Example 2:

you have one grab card, opponent has one grab card
correct grab card is on opponent's

you grabbed opponent's one, opponent does not do nothing. You win, opponent loses this round.

if you say `numOppoGrabCards -= 1` then this is not correct, this means opponent wins the game. This is not the case.

Correct transfer: `numYourGrabCards -= 1`  ----> `numYourGrabCards = 0` ---> move to `END_GAME`